api-prep/expectations.md

```diff
@@ -0,0 +1,89 @@
+# API Expectations
+
+For our API, we expect users to be able to perform a nest of
+operations to interact with our application and our API, to
+consume it and to retrieve the data that they desire.
+
+So to keep this all relative and together, this new directory
+has been created called 'api-prep' for any information to be
+referred back to at a later date. This directory will not be 
+in the final product, only in development temporarily.
+
+Starting off with an example verb/action combination, of a task
+a user might be expected to perform.
+
+> **verb**  
+> action 
+
+Here is our simple JSON structure:
+
+```.json
+{
+    "users": [
+        {
+            "user_id": 1,
+            "user_name": "Lee",
+            "anime": "Yu Yu Hakusho",
+            "anime_eps": 99,
+            "eps_desired": 99,
+            "eps_watched": 42
+        },
+        {
+            "user_id": 2,
+            "user_name": "Chelly",
+            "anime": "Sailor Moon",
+            "anime_eps": 200,
+            "eps_desired": 50,
+            "eps_watched": 150
+        }
+    ]
+}
+```
+
+This formatting applies throughout these expectations.
+
+> **POST**  
+> Update the current amount of episodes watched.
+
+We might have an API decorated method in our project called "update_user" and we can send a JSON payload in a raw format
+to an address.
+
+Example: `http://127.0.0.1:5000/api/update_user`
+
+The payload could look like:
+
+```.json
+{
+  "users": [
+    {
+      "user_id": 1,
```

Comment:
Another thing to consider... IN THE FUTURE is who should have permissions and access to change these values? Can user 1 change user 2(obviously no) so would you. Ever need the user to supply the ID? Or would you inherently already have that ID from them "signing in" to your site ? Or are these API calls backend access only so that no randos can call it. Whatever decision is made will influence how you want to build this.

Reply:
we will always have a user_id because we do not ever want to be referencing user names as ways to grab info out of the db. user_ids are faster! (whole numbers always easy to calculate ;) ) 

That being said, if we had some sort of `/auth` system for people to log in with, then the website/software/whatever would already have that user ID in context and would just pass it along. That being said, the user ID can be submitted via payload or via uri... we can discuss this more